### PR TITLE
Mark CDC as GA

### PR DESCRIPTION
**Pull request by haaawk**
CDC is ready to be a non-experimental feature so remove the experimental flag for it.
Also, guard Alternator Streams with their own experimental flag. Previously, they were using CDC experimental flag as they depend on CDC.

Tests: unit(dev)

### CODE REVIEW

**Comment by avikivity** [comment]
Dequeued. @bhalevy reports that it fails in dtest murmur3_partitioner_test.

**Comment by slivne** [comment]
@haaawk ping

File: cdc/generation.cc [hunk 1]
```diff
@@ -310,7 +310,14 @@ db_clock::time_point make_new_cdc_generation(
     auto ts = db_clock::now() + (
             (for_testing || ring_delay == milliseconds(0)) ? milliseconds(0) : (
                 2 * ring_delay + duration_cast<milliseconds>(generation_leeway)));
-    sys_dist_ks.insert_cdc_topology_description(ts, std::move(gen), { tm.count_normal_token_owners() }).get();
+    try {
+        sys_dist_ks.insert_cdc_topology_description(ts, std::move(gen), { tm.count_normal_token_owners() }).get();
+    } catch (...) {
```
Comment (kbr-) [inline cdc/generation.cc:315]: We only want to make this optional in the replace-node or rolling-upgrade case, but not in the node-bootstrap case. For node bootstrap changing the generation is mandatory.

Comment (kbr-) [reply cdc/generation.cc:315]: Hence I think this `try`+`catch` should be outside the function, in the `if (!_cdc_streams_ts) {` path (line 552 in storage service).

File: service/storage_service.cc [hunk 2]
```diff
@@ -892,24 +897,19 @@ void storage_service::bootstrap() {
...
-                    _sys_dist_ks.local(), get_ring_delay(), _for_testing);
-        } else {
-            // We should not be able to join the cluster if other nodes support CDC but we don't.
-            // The check should have been made somewhere in prepare_to_join (`check_knows_remote_features`).
-            assert(!_feature_service.cluster_supports_cdc());
-        }
+        _cdc_streams_ts = cdc::make_new_cdc_generation(db().local().get_config(),
+                _bootstrap_tokens, _token_metadata, _gossiper,
+                _sys_dist_ks.local(), get_ring_delay(), _for_testing);
+        assert(_cdc_streams_ts);
```
Comment (kbr-) [inline service/storage_service.cc:912]: This assert may fire in normal circumstances (not due to a bug), so it should not be an assert.

Comment (kbr-) [reply service/storage_service.cc:912]: Instead, the node should fail bootstrapping with a message. We should just let the exception from `make_new_cdc_generation` fly out in this case.

File: cdc/generation.cc (hunk 1 above)
Comment (haaawk) [reply cdc/generation.cc]: done

File: service/storage_service.cc (hunk 2 above)
Comment (haaawk) [reply service/storage_service.cc]: done

**Comment by haaawk** [comment]
I fixed @kbr- comments and the failing partitioner test.
BYO is running here: https://jenkins.scylladb.com/job/scylla-master/job/byo/job/byo_build_tests_dtest/1065/

Nothing changed in the previous version. Only a new commit was added to fix the failing test.

**Comment by avikivity** [comment]
@nyh since you shepherded the work maybe you should have the honor of committing the GA series.

**Comment by haaawk** [comment]
Please wait for BYO to finish though.

File: test/boost/cdc_test.cc [hunk 3]
```diff
@@ -46,9 +46,6 @@ static cql_test_config mk_cdc_test_config() {
     auto ext = std::make_shared<db::extensions>();
```
Comment (kbr-) [inline test/boost/cdc_test.cc:46]: This function should not be necessary at all now.
It was introduced in b130b76274fe7414b01ada9b41554c78540454e1 to speed up tests.
(I don't recall if the slowdown in startup due to enabling CDC wasn't fixed at some point)

Comment (haaawk) [reply test/boost/cdc_test.cc:46]: Sounds great but also deserves a PR on its own ;).

**Comment by haaawk** [comment]
First BYO failed due to some flaky dtest (migration_test.py:TTLWithMigrate.big_table_with_ttls_test). Here's another run -> https://jenkins.scylladb.com/job/scylla-master/job/byo/job/byo_build_tests_dtest/1067/

**Comment by haaawk** [comment]
All BYO tests have passed. @nyh @avikivity would you mind merging?

**Comment by avikivity** [comment]
I'll leave it to @nyh who shepherded the whole process.

**Pull request closed without merging**